### PR TITLE
Feature/unit testing ci inegration

### PR DESCRIPTION
**Pull request by KrishKittur**
No changes to code that runs on the car in this pr; just added unit testing, ci integration, and automatic doc generation.

### CODE REVIEW

File: flake.nix [hunk 1]
```diff
@@ -106,6 +106,21 @@
                 pkgs.drivebrain_software
               ];
             };
+
+            devShells.tests = pkgs.mkShell rec {
+              name = "test-devshell";
+              shellHook = ''
+                  export PLUH="hello"
```
Comment (RCMast3r) [inline flake.nix:113]: prob dont need this shellhook for just this variable. prob will need shellhook for the generated dbc file tho

File: flake.nix [hunk 2]
```diff
@@ -106,6 +106,21 @@
...
+                '';
+
+              nativeBuiltInputs = [ pkgs.drivebrain_core_msgs_proto_cpp ];
```
Comment (RCMast3r) [inline flake.nix:116]: not needed?

File: flake.nix [hunk 3]
```diff
@@ -114,8 +129,8 @@
                   easy_cmake.overlays.default
                   (final: _: { drivebrain_software = final.callPackage ./default.nix { }; })
                 ]
-                ++ data_acq.overlays.x86_64-linux
-                ++ self.flake.overlays;
+                ++ data_acq.overlays.x86_64-linux;
+                # ++ self.flake.overlays;
```
Comment (RCMast3r) [inline flake.nix:133]: remove commented code

File: drivebrain_core_impl/drivebrain_control/src/SimpleController.cpp [hunk 4]
```diff
@@ -39,7 +34,7 @@ bool control::SimpleController::init()
     std::optional max_regen_torque = get_live_parameter<torque_nm>("max_regen_torque");
     std::optional rear_torque_scale = get_live_parameter<float>("rear_torque_scale");
     std::optional regen_torque_scale = get_live_parameter<float>("regen_torque_scale");
-    std::optional positive_speed_set = get_live_parameter<speed_m_s>("positive_speed_set");
+    std::optional positive_speed_set = get_parameter_value<speed_m_s>("positive_speed_set");
```
Comment (RCMast3r) [inline drivebrain_core_impl/drivebrain_control/src/SimpleController.cpp:37]: why is this no longer a live parameter?

**Review by RCMast3r — CHANGES_REQUESTED** [review]

File: drivebrain_core_impl/drivebrain_control/src/SimpleController.cpp (hunk 4 above)
Comment (KrishKittur) [reply drivebrain_core_impl/drivebrain_control/src/SimpleController.cpp:37]: When I was creating the unit test I thought you told me to make the speed_set (which just refers to the max) constant.

Comment (RCMast3r) [reply drivebrain_core_impl/drivebrain_control/src/SimpleController.cpp:37]: Oh, well can we put it back to live, we often use this live now

File: drivebrain_core_impl/drivebrain_control/src/SimpleController.cpp
Comment (RCMast3r) [inline drivebrain_core_impl/drivebrain_control/src/SimpleController.cpp:1]: this needs to be added back

**Review by RCMast3r — CHANGES_REQUESTED** [review]

**Review by RCMast3r — APPROVED** [review]

**Review by SreeDan — APPROVED** [review]

**Pull request merged**